### PR TITLE
Remove submit button warning from Gamelab projects

### PR DESCRIPTION
**Pull request by Nokondi**
Disables confirmation dialog that appears on Gamelab levels when student presses "Submit" after running their project. Original dialog was inaccurate, and there was concern that this was discouraging students from submitting assignments.

Original task included adding a warning before students attempted to navigate away from the page without submitting their work, but the `onbeforeunload `dialog behavior was changed in 2016 to prevent setting custom messages in dialogs that prevent the user from exiting a page. After discussion, it was decided to only disable the original dialog.

Original dialog:
image

## Links
[AITT-738](https://codedotorg.atlassian.net/browse/AITT-738)

## Testing story

- Updated gamelab unit tests for submit dialog removal
- Added new UI test step to account for submit dialog removal

## Follow-up work

Currently there is a pause after a student submits, and it doesn't look like anything is happening. It seems like a good idea to add a dialog pop up to let students know to wait while their project is submitted.

Follow up task:
[AITT-800](https://codedotorg.atlassian.net/browse/AITT-800): add spinner / dialog to warn students not to navigate away while project is being submitted

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/test/ui/features/step_definitions/steps.rb [hunk 1]
```diff
@@ -1200,6 +1200,15 @@ def browser_request(url:, method: 'GET', headers: {}, body: nil, code: 200, trie
   GHERKIN
 end
 
+And(/^I submit this gamelab level$/) do
+  steps <<~GHERKIN
+    And I press "runButton"
+    And I wait to see "#submitButton"
+    And I press "submitButton"
+    And I wait for 10 seconds
```
Comment (Nokondi) [inline dashboard/test/ui/features/step_definitions/steps.rb]: I used "wait for x seconds" here to allow enough time for the project to be submitted.

Comment (davidsbailey) [reply dashboard/test/ui/features/step_definitions/steps.rb]: please use `And I press "submitButton" to load a new page` instead

**Review by davidsbailey — COMMENTED** [review]
please add Jira items to track the following work:
* add a dialog pop up to let students know to wait while their project is submitted
* remove dead code which was disabled in this PR

**Comment by Nokondi** [comment]
> please use And I press "submitButton" to load a new page instead
That's a much better solution. Thanks!

File: apps/src/submitHelper.js [hunk 2]
```diff
@@ -59,11 +61,16 @@ export function onSubmitComplete(response) {
  * When submit button is pressed, confirm, and then do it.
  */
 function onPuzzleSubmit() {
-  showConfirmationDialog({
-    title: commonMsg.submitYourProject(),
-    text: commonMsg.submitYourProjectConfirm(),
-    onConfirm: () => onPuzzleComplete(true),
-  });
+  console.log(studioApp);
```
Comment (davidsbailey) [inline apps/src/submitHelper.js]: remove debug code

File: apps/src/submitHelper.js [hunk 3]
```diff
@@ -15,6 +15,8 @@ import DialogButtons from './legacySharedComponents/DialogButtons';
 // Parameters provided by the calling app.
 let studioApp, onPuzzleComplete, unsubmitUrl;
 
+// Const to temporary disable pop up stating that student cannot resubmit assignment
+
```
Comment (davidsbailey) [inline apps/src/submitHelper.js]: remove outdated comment

File: dashboard/test/ui/features/star_labs/gamelab_submittable.feature [hunk 4]
```diff
@@ -8,7 +8,7 @@ Background:
 
 Scenario: Submit anything, unsubmit, be able to resubmit.
   # First, submit something.
-  When I submit this level
+  When I submit this gamelab level
```
Comment (davidsbailey) [inline dashboard/test/ui/features/star_labs/gamelab_submittable.feature:11]: nice solution!

**Review by davidsbailey — COMMENTED** [review]
Hey Mark, can you please:
* update the PR description to match what this PR is now doing
* include a link to a Jira ticket for this follow-up work:

> It seems like a good idea to add a dialog pop up to let students know to wait while their project is submitted.

**Comment by davidsbailey** [comment]
apologies Mark, I must have missed when you added the Jira items to the PR description. if there's any work left to do for this one can you please clarify what that is?
* [AITT-801](https://codedotorg.atlassian.net/browse/AITT-801): remove dead code

**Review by davidsbailey — APPROVED** [review]

**Pull request merged**